src/transaction/locator_cl.c

```diff
@@ -6952,3 +7031,54 @@ locator_can_skip_fetch_from_server (MOP mop, LOCK * lock, LC_FETCH_VERSION_TYPE
   /* We are here because we need to upgrade lock on object. */
   return false;
 }
+
+/*
+ * locator_lob_create_or_remove_dir() - Unified interface for creating or removing LOB directories.
+ *   return: error code
+ *   old_hfid(in): HFID of an existing LOB directory to remove.
+ *   new_hfid(in): HFID for the new LOB directory to create.
+ *   lob_attrid_arr(in): Array of LOB attribute IDs used for create/remove operations.
+ *   lob_attrid_arr_length(in): Number of elements in lob_attrid_arr.
+ *
+ * NOTE: This function abstracts the logic of calling lob_create_dir() and lob_remove_dir(),
+ *       allowing the caller to handle both operations through a single interface.
+ */
+int
+locator_lob_create_or_remove_dir (HFID * old_hfid, HFID * new_hfid, int *lob_attrid_arr, int lob_attrid_arr_length)
+{
+  int error = NO_ERROR;
+
+  assert (old_hfid != NULL || new_hfid != NULL);
+
+  if (old_hfid != NULL && new_hfid != NULL)	/* truncate case */
+    {
+      error = lob_remove_dir (old_hfid, -1);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+      error = lob_create_dir (new_hfid, lob_attrid_arr, lob_attrid_arr_length);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+    }
+  else if (old_hfid != NULL)
+    {
+      error = lob_remove_dir (old_hfid, lob_attrid_arr[0]);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+    }
+  else if (new_hfid != NULL)
+    {
+      error = lob_create_dir (new_hfid, lob_attrid_arr, lob_attrid_arr_length);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+    }
```

Comment:
아래와 같이 코드를 줄일 수 있습니다. assert() 검사도 각 if 안에서 필요한 부분만 체크하면 되겠네요.
```suggestion
  if (old_hfid != NULL)
    {
      error = lob_remove_dir (old_hfid, lob_attrid_arr[0]);
      if (error != NO_ERROR)
	{
	  return error;
	}
    }

  if (new_hfid != NULL)
    {
      error = lob_create_dir (new_hfid, lob_attrid_arr, lob_attrid_arr_length);
      if (error != NO_ERROR)
	{
	  return error;
	}
    }
```